Source/FakeItEasy.Tests/Creation/DefaultFakeCreatorTests.cs

```diff
@@ -114,6 +120,29 @@ public void CreateFake_should_pass_options_builder_that_returns_itself_for_any_c
             Assert.That(call.Invoke(builderPassedToAction), Is.SameAs(builderPassedToAction));
         }
 
+        [TestCaseSource("invalidImplementsCalls")]
+        public void CreateFake_should_throw_when_type_passed_to_implements_is_not_an_interface(Func<IFakeOptionsBuilder<Foo>, IFakeOptionsBuilder<Foo>> call)
+        {
+            Guard.AgainstNull(call, "call");
+
+            // Arrange
+            IFakeOptionsBuilder<Foo> builderPassedToAction = null;
+
+            // Act
+            this.creator.CreateFake<Foo>(x => { builderPassedToAction = x; });
+
+            // Assert
+            Assert.Throws<ArgumentException>(() => call.Invoke(builderPassedToAction));
+        }
+
+        [Test]
+        public void CreateFake_implements_should_be_null_guarded()
+        {
+            // Arrange, Act, Assert
+            NullGuardedConstraint.Assert(() => this.creator.CreateFake<IFoo>(x => x.Implements(null)));
+            NullGuardedConstraint.Assert(() => this.creator.CreateFake<IFoo>(x => x.Wrapping(A.Fake<IFoo>()).Implements(null)));
+        }

```

Comment:
@jimmyheaddon, I like the new test, but while I'm complaining about small things (:wink:), this is sort of testing two things (ultimately the same implementation, but we don't know that from where we stand).
Would you consider splitting it in two? Maybe the new test could be `CreateFake_implements_after_wrapping_should_be_null_guarded` or similar?
Otherwise, if the first Assert fails, we'll never know what happens with the second one…